sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/BufferedPublisher.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Messaging.EventHubs.Tests;
+
+namespace Azure.Messaging.EventHubs.Stress
+{
+    internal class BufferedPublisher
+    {
+        private string connectionString;
+        private string eventHubName;
+        private Metrics metrics;
+        private ProducerConfiguration testConfiguration;
+
+        public BufferedPublisher(ProducerConfiguration testConfigurationIn, Metrics metricsIn)
+        {
+            connectionString = testConfigurationIn.EventHubsConnectionString;
+            eventHubName = testConfigurationIn.EventHub;
+            testConfiguration = testConfigurationIn;
+            metrics = metricsIn;
+        }
+
+        public async Task Start(CancellationToken cancellationToken)
+        {
+            var enqueueTasks = new List<Task>();
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                using var backgroundCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+                try
+                {
+                    var producer = new EventHubBufferedProducerClient(connectionString, eventHubName);
+                    producer.SendEventBatchSucceededAsync += args =>
+                    {
+                        var numEvents = args.EventBatch.ToList().Count;
+                        var eventProperties = new Dictionary<String, String>();
+
+                        metrics.Client.GetMetric(metrics.SuccessfullySentFromQueue, args.PartitionId).TrackValue(numEvents);
+
+                        return Task.CompletedTask;
+                    };
+
+                    producer.SendEventBatchFailedAsync += args =>
+                    {
+                        var numEvents = args.EventBatch.ToList().Count;
+
+                        var eventProperties = new Dictionary<String, String>();
+
+                        metrics.Client.GetMetric(metrics.EventsNotSentAfterEnqueue, args.PartitionId).TrackValue(numEvents);
+                        metrics.Client.TrackException(args.Exception);
+
+                        return Task.CompletedTask;
+                    };
+
+                    await using (producer.ConfigureAwait(false))
```

Comment:
ooohhh... I see what you're up to.    Clever idea.   We should probably consider being more explicit about the lifecycle, though.   It would be easy for someone to miss this if they weren't familiar.